gluten-core/src/main/java/org/apache/gluten/substrait/expression/ExpressionBuilder.java

```diff
@@ -215,6 +217,19 @@ public static LiteralNode makeLiteral(Object obj, TypeNode typeNode) {
 
   public static LiteralNode makeLiteral(Object obj, DataType dataType, Boolean nullable) {
     TypeNode typeNode = ConverterUtils.getTypeNode(dataType, nullable);
+    if (obj instanceof UnsafeArrayData) {
```

Comment:
@gaoyangxiaozhu, can we just overload `makeLiteral` with `UnsafeArrayData` type as input?

Reply:
em.. actually not sure, unless we sure the literal list type always be `UnSafeArrayData`, but i am not sure right now.
Let's keep current change , and let me do follow up to check spark code to see if can always move to `UnSafeArrayData`